backend/module/chaos/serverexperimentation/xds/xds_cache.go

```diff
@@ -77,36 +82,59 @@ func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCa
 
 	// Create/Update experiments
 	for cluster, experiments := range clusterFaultMap {
+		resources := make(map[gcpTypes.ResponseType][]gcpTypes.ResourceWithTtl)
+
 		logger.Debugw("Injecting fault for cluster", "cluster", cluster)
 
-		runtimeResource := generateRTDSResource(experiments, rtdsConfig, ttl, logger)
-		err := setSnapshot(runtimeResource, cluster, snapshotCache, logger)
+		// Enable ECDS if cluster present in the map
+		if _, exists := ecdsConfig.enabledClusters[cluster]; exists {
+			resources[gcpTypes.ExtensionConfig] = generateECDSResource(experiments, ttl, logger)
+
+			// generate empty Runtime config to clear any existing RTDS faults
+			resources[gcpTypes.Runtime] = generateRTDSResource([]*experimentation.Experiment{}, rtdsConfig, ttl, logger)
```

Comment:
Seems like we don't actually implement the logic for enabling this for everything? That's probably okay for now though we'll want to update the proto docs in that case